src/components/Grades.js

```diff
@@ -27,106 +27,96 @@ const useStyles = makeStyles((theme) => ({
   classHeaderSpan: {
     fontWeight: 600,
     color: 'rgba(0, 0, 0, 0.75)',
-    fontSize: 20,
-  },
-  content: {
-    paddingTop: 0,
-    overflowX: 'scroll',
+    fontSize: 16,
   },
   tableCell: {
     color: 'rgba(0, 0, 0, 1)',
     fontWeight: 600,
     fontSize: 14,
     width: '33%',
   },
+  empty: {
+    textAlign: 'center',
+  },
+  mobile: {
+    overflowX: 'scroll',
+  },
 }))
 
-const GradeRow = (courses) => {
-  try {
-    let tableArray = []
-    for (let i = 0; i < courses.length; i++) {
-      let course = courses[i].subjectCode + ' - ' + courses[i].subjectNumber
-      let credits = courses[i].grade.credit
-      let grade = courses[i].grade.grade
-
-      tableArray.push(
-        <TableRow key={i + Math.random()}>
-          <TableCell>{course}</TableCell>
-          <TableCell>{credits}</TableCell>
-          <TableCell>{grade}</TableCell>
-        </TableRow>
-      )
-    }
-    return tableArray
-  } catch (error) {
-    return <TableRow />
-  }
+const CourseGrades = ({ courses }) => {
+  return courses.map((course, i) => {
+    return (
+      <TableRow key={i}>
+        <TableCell>
+          {course.subjectCode} - {course.subjectNumber}
+        </TableCell>
+        <TableCell>{course.grade.credit}</TableCell>
+        <TableCell>{course.grade.grade}</TableCell>
+      </TableRow>
+    )
+  })
 }
 
-const getOverallCredits = (credits) => {
-  let rows = []
-  let i = 0
-  for (let cr of credits) {
-    rows.push(
-      <TableRow key={i++}>
-        <TableCell>{cr.standing}</TableCell>
-        <TableCell>{cr.credits}</TableCell>
-        <TableCell>{cr.gpa}</TableCell>
+const Credits = ({ credits }) => {
+  return credits.map((credit, i) => {
+    return (
+      <TableRow key={i}>
+        <TableCell>{credit.standing}</TableCell>
+        <TableCell>{credit.credits}</TableCell>
+        <TableCell>{credit.gpa}</TableCell>
       </TableRow>
     )
-  }
-  return rows
+  })
 }
 
 const Grades = (props) => {
   const classes = useStyles()
-  const { mobile } = props
   const credits = useSelector((state) => state.credits)
   const courses = useSelector((state) => state.courses)
-  const courses_fetched = useSelector((state) => state.fetched)
   const courses_error = useSelector((state) => state.error)
+  const { mobile } = props
 
-  if (courses_error || credits.length === 0) {
+  if (courses_error) {
     return (
       <div className={classes.error}>
         <ErrorMessages />
       </div>
     )
-  }
-
-  if (credits !== []) {
+  } else if (courses.length === 0 && credits.length === 0) {
+    return <Typography className={classes.empty}>No data to display</Typography>
+  } else {
```

Comment:
Can you have credits for a term but not have any courses? Otherwise could this just be checking the `courses` array length?

Reply:
Theoretically, you could have credits but no courses if the first fetch was successful and any thereafter weren't, because credits are only fetched once. But I think that if there's an error with the fetch I just display an error message anyway lol. So I'll clean up this check